tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -230,7 +230,7 @@ class Restrictions {
     if (!config.isExperimentalFeatureEnabled(ExperimentalFeature.inheritance)) {
       return [
         SourceSpanSeverityException(
-          'The "${Keyword.extendsClass}" key can only be used when the (experimental) inheritance feature is enabled.',
+          'The "${Keyword.extendsClass}" property is not allowed for class type. Valid keys are {class, extends, table, managedMigration, serverOnly, fields, indexes}.',
```

Comment:
Suggestion: I think we could create a generic solution for this case instead that should be able to print this list dynamically. 

If we add a new optional field to the `ValidateNode` class called `isDisabled` as a flag to enable a keyword (set default to false).

Then we can add a filter for this in the method `_collectInvalidKeyErrors`, like so:

```dart
var validKeys =
      documentStructure.where((e) => !e.isDisabled).map((e) => e.key).toSet();
```
This is the global method to check for valid keys and will work recursively for any level in the node tree.

Then in the class_yaml_definition you just need to set the isEnabled variable from the call to `config.isExperimentalFeatureEnabled(ExperimentalFeature.inheritance)`.

E.g.

```dart
ValidateNode(
    Keyword.extendsClass,
    isDisabled: !restrictions.config.isExperimentalFeatureEnabled(ExperimentalFeature.inheritance)
    ...
),
```

If you leave the test as is it should pass with the above modification.

Reply:
On second thought perhaps `isHidden` is a better word to use.